src/diffusers/models/modeling_utils.py

```diff
@@ -719,9 +719,10 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
 
         if hf_quantizer is not None:
             is_bnb_quantization_method = hf_quantizer.quantization_config.quant_method.value == "bitsandbytes"
-            if is_bnb_quantization_method and device_map is not None:
+            is_torchao_quantization_method = hf_quantizer.quantization_config.quant_method.value == "torchao"
```

Comment:
I would just simplify and error out if `device_map` is not None.

Reply:
Updated